runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorkerTest.java

```diff
@@ -2916,6 +2916,69 @@ public void testActiveThreadMetric() throws Exception {
     executor.shutdown();
   }
 
+  @Test
+  public void testOverrideMaximumThreadCount() throws Exception {
+    int maxThreads = 5;
+    int threadExpirationSec = 60;
+    CountDownLatch processStart1 = new CountDownLatch(2);
+    AtomicBoolean stop = new AtomicBoolean(false);
+    // setting up actual implementation of executor instead of mocking to keep track of
+    // active thread count.
+    BoundedQueueExecutor executor =
```

Comment:
the test that BoundedQueueExecutor works when calling setMaximumPoolSize seems like it should be in BoundedQueueExecutorTest.

The test at the streaming dataflow worker level seems like it should have the work update client and validate plumbing of responses to setting the max on the queue.  The queue itself could be a mock in that test.

Reply:
Sounds good. Will refactor it.